app/models/ingredient.rb

```diff
@@ -1,3 +1,14 @@
 class Ingredient < ApplicationRecord
    validates_presence_of :name, :cost
+
+   has_many :recipe_ingredients
+   has_many :recipes, through: :recipe_ingredients
+
+   def self.total_cost
+      self.sum(:cost)
+   end
```

Comment:
I interpreted US 2 as saying we want the total cost of ingredients in a specific recipe meaning I'd make this an instance method on the `Recipe` model. You're calculating the total cost of ALL ingredients regardless of recipe. You are using active record correctly, however, so that's great! If you moved this to `Recipe`, it would look something like this:
```ruby
class Recipe < Application Record
  ...relationships and validations...

  def total_cost
    ingredients.sum(:cost)
  end
```